### PR TITLE
Collection + Relations v1.

### PR DESCRIPTION
**Pull request by isacandrei**

### CODE REVIEW

File: src/Helpers/Collection.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+ * Class Collection
+ * @package Joomla\Entity\Helpers
+ * @since   1.0
+ */
+class Collection implements ArrayAccess, IteratorAggregate, JsonSerializable
+{
+	/**
+	 * The items contained in the collection.
+	 *
+	 * @var array
```
Comment (wilsonge) [inline src/Helpers/Collection.php]: `Model[]` instead of `array`

File: src/Query.php [hunk 2]
```diff
@@ -215,7 +216,7 @@ public function findLast($columns = array('*'))
 		$item = $this->db->setQuery($this->query)->loadAssoc();
 
 		// TODO something like first() from Collection would make this nicer
```
Comment (wilsonge) [inline src/Query.php]: Remove the TODO :)

**Comment by george-wilson-iproov** [comment]
We need tests for the collection class :)

File: src/Query.php [hunk 3]
```diff
@@ -226,7 +227,7 @@ public function findLast($columns = array('*'))
 	 */
 	public function whereKey($id)
 	{
-		$this->query->where($this->model->getPrimaryKey() . ' = ' . $id);
```
Comment (george-wilson-iproov) [inline src/Query.php]: What's the setLimit here for? Feels less intitutive

Comment (isacandrei) [reply src/Query.php]: The primary key should be unique anyway right? So the setLimit is not really a necessity?

Comment (wilsonge) [reply src/Query.php]: Correct

File: src/Query.php [hunk 4]
```diff
@@ -188,7 +189,6 @@ public function find($id, $columns = array('*'))
 
 		$item = $this->db->setQuery($this->query)->loadAssoc();
 
-		// TODO something like first() from Collection would make this nicer
 		return $this->hydrate(array($item))[0];
```
Comment (wilsonge) [inline src/Query.php]: You're missing the `first()` call here

**Pull request merged**